### PR TITLE
Moving ETag to ::Azure

### PR DESCRIPTION
**Pull request by vhvb1989**
Depends on merging first: https://github.com/Azure/azure-sdk-for-cpp/pull/1800

Moves ETag to Azure::

As requested in: #1789

### CODE REVIEW

File: sdk/core/azure-core/CMakeLists.txt [hunk 1]
```diff
@@ -78,10 +77,11 @@ set(
     inc/azure/core/operation.hpp
     inc/azure/core/operation_status.hpp
     inc/azure/core/platform.hpp
-    inc/azure/core/response.hpp
     inc/azure/core/uuid.hpp
     inc/azure/core/version.hpp
     inc/azure/core.hpp
+    inc/azure/etag.hpp
```
Comment (ahsonkhan) [inline sdk/core/azure-core/CMakeLists.txt:83]: Looking at this  bit more closely, `Etag` which is just a plain wrapper on string belonging in the root Azure namespace seems too visible.
I don't think folks would be using this type all that often.

Comment (ahsonkhan) [reply sdk/core/azure-core/CMakeLists.txt:83]: But .NET has it under Azure. @tg-msft, @pakrym - what do you think? Should this be at the root `Azure` namespace?

Comment (pakrym) [reply sdk/core/azure-core/CMakeLists.txt:83]: It's something customers would construct to make client calls and in .NET these types are in `Azure` namespace.

Comment (ahsonkhan) [reply sdk/core/azure-core/CMakeLists.txt]: Really? Can you please share a couple of public APIs that accept Etag as input in .NET?
C++ currently only returns it as output, from model types.

File: sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp [hunk 2]
```diff
@@ -100,7 +100,7 @@ namespace Azure { namespace Storage { namespace Blobs {
   private:
     std::string GetResumeToken() const override
     {
-      // Not supported
+      // Not sud
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/blob_responses.hpp:103]: Not sure what caused this change.
```suggestion
      // Not supported
```

**Review by RickWinter — APPROVED** [review]

**Review by ahsonkhan — APPROVED** [review]

File: sdk/storage/ci.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
```
Comment (ahsonkhan) [inline sdk/storage/ci.yml:1]: @danieljurek  is there a way to validate in CI that the yml file being modified is correct, so it fails CI (without having to trigger this pipeline manually).

**Comment by JeffreyRichter** [comment]
I think this should be in the Http namespace - it is HTTP-specific.

**Pull request merged**